Small tracking UI improvements

mateiz opened the pull request:
This PR makes a few small fixes:

1) Got rid of NaNs shown on the Compare Runs page for run IDs that didn't have a specific metric.
2) Changed the collapse/expand button for opening the Experiments sidebar to have `cursor: pointer` so it looks like a clickable object.
3) Added a breadcrumb link back to each experiment on the Run, Compare Runs and Metrics pages (the "Default > ..." in the headings below):

![screen shot 2018-08-07 at 3 52 05 pm](https://user-images.githubusercontent.com/228859/43806814-63bbf14c-9a5a-11e8-960e-4a24eee1aefc.png)

![screen shot 2018-08-07 at 3 51 47 pm](https://user-images.githubusercontent.com/228859/43806820-66ddad70-9a5a-11e8-997a-4d39ffdb4503.png)

![screen shot 2018-08-07 at 6 52 16 pm](https://user-images.githubusercontent.com/228859/43811765-173593b4-9a73-11e8-9398-32a19dfee5ba.png)

mateiz committed:
A few UI tweaks

edits

fixes

breadcrumbs for comparing metrics too

gioa commented:
The breadcrumb looks good to me! 

Nits: 
- For .breadcrumb-chevron, can we use 4px instead of 3px? ([a good practice ](https://builttoadapt.io/intro-to-the-8-point-grid-system-d2573cde8632)to have 8-pt based spacing). 
- In the two mockups, the spacing between the page title and the metadata is inconsistent and the second one (Comparing page) spacing looks right.

mateiz commented:
Thanks Joy. I've fixed those except that the space around the caret looks unbalanced with 8 points on both sides, perhaps due to the font, so I added slightly more space on the left.

mateiz committed:
more fixes

andrewmchen commented on mlflow/server/js/src/components/CompareRunView.js:
```diff
@@ -31,22 +34,34 @@ class CompareRunView extends Component {
       <div className="CompareRunView">
-        <h1>Comparing {this.props.runInfos.length} Runs</h1>
+        {experiment ?
+            <h1>
```
Any thoughts about combining this breadcrumb and the other one and using a shared `Breadcrumb` component?

andrewmchen commented on mlflow/server/js/src/components/CompareRunView.js:
```diff
@@ -31,22 +34,34 @@ class CompareRunView extends Component {
       },
     };
 
+    let experiment = this.props.experiment;
```
nit: let -> const?

andrewmchen commented on mlflow/server/js/src/components/CompareRunView.js:
```diff
@@ -1,9 +1,9 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { getParams, getRunInfo } from '../reducers/Reducers';
+import {getExperiment, getParams, getRunInfo} from '../reducers/Reducers';
```
nit: `{getExperiment, getParams, getRunInfo}` -> `{ getExperiment, getParams, getRunInfo }`. We should probably have a standard way of setting up common IDEs.

andrewmchen commented on mlflow/server/js/src/components/CompareRunView.js:
```diff
@@ -13,6 +13,9 @@ import { getLatestMetrics } from '../reducers/MetricReducer';
 
 class CompareRunView extends Component {
   static propTypes = {
+    // This prop is optional in case we allow comparing runs across experiments in the future
```
As discussed offline. It might be simpler to have this be required for now and then add back the code which allows you to compare across experiments.

mateiz committed:
review comments and other updates

mateiz committed:
small fix

mateiz committed:
spacing fix

mateiz committed:
style
